specification/monitor/resource-manager/Microsoft.Insights/preview/2019-10-17-preview/privateLinkScopes_API.json

```diff
@@ -40,17 +43,17 @@
         "operationId": "PrivateLinkScopes_List",
         "parameters": [
           {
-            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/ApiVersionParameter"
+            "$ref": "#/parameters/ApiVersionParameter"
           },
           {
-            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/SubscriptionIdParameter"
+            "$ref": "#/parameters/SubscriptionIdParameter"
           }
         ],
         "responses": {
           "default": {
             "description": "Error response describing why the operation failed.",
             "schema": {
-              "$ref": "../../../../../common-types/resource-management/v1/types.json#/definitions/ErrorResponse"
+              "$ref": "#/definitions/ErrorResponseCommon"
```

Comment:
>ErrorResponseCommon [](start = 37, length = 19)

ErrorResponseCommon declares a response with code, message, etc. properties which is not correct. Take a look at [https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/common-api-details.md#error-response-content](https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/common-api-details.md#error-response-content). The top-level property should be `error`, which is an object that contains `code`, `message`, and other properties.

Reply:
this definition of ErrorResponseCommon was copied from https://github.com/Azure/azure-rest-api-specs/blob/master/specification/common-types/resource-management/v1/types.json#L272

Can you confirm whether this definition need to be updated?

Reply:
No, not exactly. The issue is that you are making your error response like this:
```
{
  code: "",
  message: ""
}
```
It needs to be like this:
```
{
  error: {
    code: "",
    message: "",
    ....
  }
}
```

So the definition in common-types is correct if you use it as the schema of the "error" property.

---
In reply to: [401315210](https://github.com/Azure/azure-rest-api-specs/pull/8819#discussion_r401315210) [](ancestors = 401315210)

Reply:
marcin you are **wrong**. 

Please see documentation for azure async operation response
[https://docs.microsoft.com/en-us/azure/azure-resource-manager/management/async-operations#azure-asyncoperation-request-and-response](url)

This is the right contract:
```

{
    "id": "{resource path from GET operation}",
    "name": "{operation-id}", 
    "status" : "Succeeded | Failed | Canceled | {resource provider values}", 
    "startTime": "2017-01-06T20:56:36.002812+00:00",
    "endTime": "2017-01-06T20:56:56.002812+00:00",
    "percentComplete": {double between 0 and 100 },
    "properties": {
        /* Specific resource provider values for successful operations */
    },
    "error" : { 
        "code": "{error code}",  
        "message": "{error description}" 
    }
}

```
what  you're suggesting is **wrong** and against the documentation. You suggested the following:
```

{
    "id": "{resource path from GET operation}",
    "name": "{operation-id}", 
    "status" : "Succeeded | Failed | Canceled | {resource provider values}", 
    "startTime": "2017-01-06T20:56:36.002812+00:00",
    "endTime": "2017-01-06T20:56:56.002812+00:00",
    "percentComplete": {double between 0 and 100 },
    "properties": {
        /* Specific resource provider values for successful operations */
    },
    "error" : { 
        "error": {
            "code": "{error code}",  
            "message": "{error description}" 
       }
    }
}

```

Please also see the master. veryealy doesn't do any breaking change here. He merely changed naming.